core/server/worker/src/main/java/alluxio/worker/grpc/AbstractReadHandler.java

```diff
@@ -100,9 +103,10 @@
   AbstractReadHandler(ExecutorService executorService,
       StreamObserver<ReadResponse> responseObserver, AuthenticatedUserInfo userInfo) {
     mDataReaderExecutor = executorService;
-    mSerializingExecutor = new SerializingExecutor(executorService);
     mResponseObserver = responseObserver;
     mUserInfo = userInfo;
+    mSerializingExecutor =
```

Comment:
Why do we need two thread pools?

Reply:
Removed  `mDataReaderExecutor ` in #10540, it looks fine.

Reply:
1. The data reader thread shouldn't be blocked by sending responses back to the client. mSerializingExecutor could block the data reader from doing I/O. Ideally we minimize latency by using a dedicated threadpool for the data readers performing I/O.
2. The serializing executor should guarantee that everything submitted to it can be run (no RejectedExecutionExceptions). In order to guarantee this, we use a cached threadpool with a max capacity of `Integer.MAX_VALUE`. The pool in my testing does not grow above 10 threads or so, but if we use the same executor as the data reader, then we could fail to send responses to the client after erroring, reading data, or even completing the read entirely because submitting the tasks in the `reply*` methods could fail.

Reply:
For context, I'm trying to snuff out issues when we hit the capacity of the `GrpcExecutors#BLOCK_READER_EXECUTOR`.

In my testing I set this to some artificially low value (32, 64, 128), and then generate a few thousand clients worth of load on a single worker. I want to make sure the system behaves well when we scale those up to 2048 threads and anywhere from 10k-50k clients, or at least fails gracefully in such situations.

Reply:
Cross-thread operations on large data blocks will cause CPU cache invalidation and decrease the throughput of worker processes.

Reply:
We can test that change in your PR. I am trying to limit the scope of this so that it is suitable for a 2.2.X release

Thanks for your for help!